File: note-keeper-service/src/main/java/xyz/crearts/note/keeper/dto/NoteInput.java
package xyz.crearts.note.keeper.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import xyz.crearts.note.keeper.model.Attachment;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class NoteInput {
    @NotBlank(message = "Title is required")
    private String title;
    private String content;
    private List<String> tags;
    private String folder;
    private String subfolder;
    private String priority;
    private Boolean isFavorite;
    private Boolean isEncrypted;
    private LocalDateTime reminder;
    private List<Attachment> attachments;
    private String templateId;
}

File: android/app/src/main/java/xyz/crearts/notekeeper/data/model/NoteInput.kt
package xyz.crearts.notekeeper.data.model

data class NoteInput(
    val title: String,
    val content: String? = null,
    val tags: List<String>? = null,
    val folder: String? = null,
    val subfolder: String? = null,
    val priority: String? = null,
    val isFavorite: Boolean? = null,
    val isEncrypted: Boolean? = null,
    val reminder: String? = null,
    val attachments: List<AttachmentInput>? = null,
    val templateId: String? = null
)

data class AttachmentInput(
    val id: String? = null,
    val name: String,
    val url: String,
    val size: Long? = null,
    val mimeType: String? = null,
    val uploadedAt: String? = null
)
